pkg/transparentproxy/config/config_executables.go

```diff
@@ -301,7 +227,22 @@ func (c InitializedExecutablesIPvX) Restore(
 	ctx context.Context,
 	rules string,
 ) (string, error) {
-	return c.restore(ctx, rules, FlagNoFlush)
+	f, err := createTempFile(c.IptablesRestore.prefix)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	defer os.Remove(f.Name())
+
+	// Log the file name and the rules to be written.
+	c.logger.Info("writing the following rules to file:", f.Name())
+	c.logger.InfoWithoutPrefix(strings.TrimSpace(rules))
+
+	if err := writeToFile(rules, f); err != nil {
+		return "", err
+	}
+
+	return c.restore(ctx, f, FlagNoFlush)
```

Comment:
Should we wrap errors in this func to have better understanding what failed if anything goes wrong?

Reply:
I don't think we should. Here is the example when I force failure:
```sh
# cleaning up any existing transparent proxy iptables rules
# [iptables] starting cleanup of existing transparent proxy rules. Any rule found in chains with names starting with "KUMA_MESH" or containing comments starting with "kuma/mesh/transparent/proxy" will be deleted
# [iptables] no transparent proxy rules detected. No cleanup necessary
# [ip6tables] starting cleanup of existing transparent proxy rules. Any rule found in chains with names starting with "KUMA_MESH" or containing comments starting with "kuma/mesh/transparent/proxy" will be deleted
# [ip6tables] no transparent proxy rules detected. No cleanup necessary
# kumactl is about to apply the iptables rules that will enable transparent proxying on the machine. The SSH connection may drop. If that happens, just reconnect again
# [iptables] writing the following rules to file: /tmp/iptables-rules.1645257655.txt
* raw
-A PREROUTING -m comment --comment "kuma/mesh/transparent/proxy/assign connection tracking zone 1 to all DNS responses" -p udp --sport 53 -j CT --zone 1
-A OUTPUT -m comment --comment "kuma/mesh/transparent/proxy/assign connection tracking zone 1 to DNS traffic from the kuma-dp user (UID 5678)" -p udp --dport 53 -m owner --uid-owner 5678 -j CT --zone 1
-A OUTPUT -m comment --comment "kuma/mesh/transparent/proxy/assign connection tracking zone 2 to DNS responses from the kuma-dp DNS proxy" -p udp --sport 15053 -m owner --uid-owner 5678 -j CT --zone 2
-A OUTPUT -m comment --comment "kuma/mesh/transparent/proxy/assign connection tracking zone 2 to all DNS requests" -p udp --dport 53 -j CT --zone 2
COMMIT
* nat
-N KUMA_MESH_INBOUND
-N KUMA_MESH_OUTBOUND
-N KUMA_MESH_INBOUND_REDIRECT
-N KUMA_MESH_OUTBOUND_REDIRECT
-A PREROUTING -m comment --comment "kuma/mesh/transparent/proxy/redirect inbound TCP traffic to our custom chain for processing" -p tcp -j KUMA_MESH_INBOUND
-I OUTPUT 1 -m comment --comment "kuma/mesh/transparent/proxy/skip further processing for configured IP addresses, ports and UIDs" -d 10.0.0.1 -j RETURN
-I OUTPUT 2 -m comment --comment "kuma/mesh/transparent/proxy/return early for DNS traffic from kuma-dp" -p udp --dport 53 -m owner --uid-owner 5678 -j RETURN
-I OUTPUT 3 -m comment --comment "kuma/mesh/transparent/proxy/redirect all DNS requests to the kuma-dp DNS proxy (listening on port 15053)" -p udp --dport 53 -j REDIRECT --to-ports 15053
-A OUTPUT -m comment --comment "kuma/mesh/transparent/proxy/redirect outbound TCP traffic to our custom chain for processing" -p tcp -j KUMA_MESH_OUTBOUND
-A KUMA_MESH_INBOUND -m comment --comment "kuma/mesh/transparent/proxy/redirect all inbound traffic to the custom chain for processing" -p tcp -j KUMA_MESH_INBOUND_REDIRECT
-A KUMA_MESH_OUTBOUND -m comment --comment "kuma/mesh/transparent/proxy/prevent traffic loops by ensuring traffic from the sidecar proxy (using 127.0.0.6/32) to loopback interface is not redirected again" -s 127.0.0.6/32 -o lo -j RETURN
-A KUMA_MESH_OUTBOUND -m comment --comment "kuma/mesh/transparent/proxy/redirect outbound TCP traffic (except to DNS port 53) destined for loopback interface, but not targeting address 127.0.0.1/32, and owned by UID 5678 (kuma-dp user) to KUMA_MESH_INBOUND_REDIRECT chain for proper handling" -p tcp ! --dport 53 -o lo ! -d 127.0.0.1/32 -m owner --uid-owner 5678 -j KUMA_MESH_INBOUND_REDIRECT
-A KUMA_MESH_OUTBOUND -m comment --comment "kuma/mesh/transparent/proxy/return outbound TCP traffic (except to DNS port 53) destined for loopback interface, owned by any UID other than 5678 (kuma-dp user)" -p tcp ! --dport 53 -o lo -m owner ! --uid-owner 5678 -j RETURN
-A KUMA_MESH_OUTBOUND -m comment --comment "kuma/mesh/transparent/proxy/return outbound traffic owned by UID 5678 (kuma-dp user)" -m owner --uid-owner 5678 -j RETURN
-A KUMA_MESH_OUTBOUND -m comment --comment "kuma/mesh/transparent/proxy/redirect all DNS requests sent via TCP to kuma-dp DNS proxy (listening on port 15053)" -p tcp --dport 53 -j REDIRECT --to-ports 15053
-A KUMA_MESH_OUTBOUND -m comment --comment "kuma/mesh/transparent/proxy/return traffic destined for localhost (127.0.0.1/32) to avoid redirection" -d 127.0.0.1/32 -j RETURN
-A KUMA_MESH_OUTBOUND -m comment --comment "kuma/mesh/transparent/proxy/redirect all other outbound traffic to our custom chain for further processing" -j KUMA_MESH_OUTBOUND_REDIRECT
-A KUMA_MESH_INBOUND_REDIRECT -m comment --comment "kuma/mesh/transparent/proxy/redirect TCP traffic to envoy (port 15006)" -p tcp -j REDIRECT --to-ports 15006
-A KUMA_MESH_OUTBOUND_REDIRECT -m comment --comment "kuma/mesh/transparent/proxy/redirect TCP traffic to envoy (port 15001)" -p tcp -j REDIRECT --to-ports 15001
COMMIT
* mangle
-A PREROUTING -m comment --comment "kuma/mesh/transparent/proxy/drop packets in the INVALID state to prevent potential issues with malformed or out-of-order packets" -m conntrack --ctstate INVALID -j DROP
COMMIT
foo
# [iptables] [1/5] iptables-nft-restore --noflush /tmp/iptables-rules.1645257655.txt
# [iptables] [1/5] restoring failed: exit status 1: iptables-nft-restore: line 31 failed
# [iptables] [1/5] will try again in 2s
# [iptables] [2/5] iptables-nft-restore --noflush /tmp/iptables-rules.1645257655.txt
# [iptables] [2/5] restoring failed: exit status 1: iptables-nft-restore: line 8 failed: Chain already exists.
# [iptables] [2/5] will try again in 2s
# [iptables] [3/5] iptables-nft-restore --noflush /tmp/iptables-rules.1645257655.txt
# [iptables] [3/5] restoring failed: exit status 1: iptables-nft-restore: line 8 failed: Chain already exists.
# [iptables] [3/5] will try again in 2s
# [iptables] [4/5] iptables-nft-restore --noflush /tmp/iptables-rules.1645257655.txt
# [iptables] [4/5] restoring failed: exit status 1: iptables-nft-restore: line 8 failed: Chain already exists.
# [iptables] [4/5] will try again in 2s
# [iptables] [5/5] iptables-nft-restore --noflush /tmp/iptables-rules.1645257655.txt
# [iptables] [5/5] restoring failed: exit status 1: iptables-nft-restore: line 8 failed: Chain already exists.
Error: failed to setup transparent proxy: unable to restore iptables rules: iptables-nft-restore failed
exit status 1
```